cmake/CPM.cmake

```diff
@@ -476,6 +502,8 @@ endfunction()
 
 # Download and add a package from source
 function(CPMAddPackage)
+  cpm_set_cmake_policies()
```

Comment:
Does it make sense to update the policies on every new package added? If we set the defaults in the beginning of the CPM script,I think it should work just as well, unless the user explicitly decides to change the default policies back.

Reply:
I wanted to have the change of the policies occur when CPM was first called instead of first included. That seemed to me when people would expect the behavior of `CPM_SET_RECOMMENDED_CMAKE_POLICIES` to be activated.

I think people would presume the following would work:

```cmake
set(CPM_SET_RECOMMENDED_CMAKE_POLICIES ON)
CPMAddPackage(....)
```

Reply:
The upside of going with the whole file policy setting approach is we could control the scope of influence to be just the CPM functions.

Currently since I am not pushing/popping policy scopes this behavior will leak to the callers CMake code.